src/ir/ty.rs

```diff
@@ -835,6 +835,22 @@ impl Type {
             _ => {}
         }
 
+        // Objective C template type parameter
+        if ty_kind == CXType_Typedef
```

Comment:
Well, we can always improve on this. Can you leave a comment with something like `// FIXME: Probably not the right thing to do.`?

Given this is an objc-only path, I believe it's fine to just merge it if it helps, so r=me with that and the style nits.